src/effects.js

```diff
@@ -153,14 +154,11 @@ function defaultPrefilter( elem, props, opts ) {
 
 		// Identify a display type, preferring old show/hide data over the CSS cascade
 		restoreDisplay = dataShow && dataShow.display;
-		if ( restoreDisplay == null ) {
-			restoreDisplay = dataPriv.get( elem, "display" );
-		}
+
 		display = jQuery.css( elem, "display" );
```

Comment:
There are certainly other problems with it, but the PR as it stands does not address the perf issue.